content-security-policy/embedded-enforcement/embedding_csp-header.sub.html

```diff
@@ -0,0 +1,109 @@
+<!DOCTYPE html>
+<html>
+<head>
+    <script src="/resources/testharness.js"></script>
+    <script src="/resources/testharnessreport.js"></script>
+    <script src="./support/testharness-helper.sub.js"></script>
+</head>
+<body>
+    <script>
+      var embedding_src = '../support/echo-embedding-csp.py';
+      var policy_src = '../support/echo-policy.py?policy=';
+      var redirect_url = '/common/redirect.py?location=';
+      var support = 'http://{{host}}:{{ports[http][0]}}/content-security-policy/support/';
+      function generateRedirect(target) { 
+        return redirect_url + support + target; 
+      }
+
+      async_test(t => {
+        var i = document.createElement('iframe');
+        i.src = embedding_src;
+
+        window.addEventListener('message', t.step_func(e => {
+          if (e.source != i.contentWindow)
+              return;
+          assert_equals(e.data['embedding_csp'], null);
+          t.done();
+        }));
+
+        document.body.appendChild(i);
```

Comment:
Several of these tests could be combined into something a little more streamlined. Perhaps two new helpers:

```
function wait_for_message_from_frame(i) {
  return new Promise((resolve, reject) => {
    window.addEventListener("message", test.step_func(e => {
      if (e.source == i.contentWindow)
        resolve(e);
    }));
  });
}

function assert_embedding_csp(test, src, csp, expected) {
  var i = document.createElement('iframe');
  if (csp !== null)
    i.csp = csp;
  i.src = src;

  wait_for_message_from_frame(i)
    .then(t.step_func_done(e => {
      assert_equals(e.data['embedding_csp'], expected);
      i.remove();
    }));

  document.body.appendChild(i);
}
```

And then tests like:

```
var tests = [
  { "src": embedding_src, "csp": null, "expected": null },
  { "src": embedding_src, "csp": "script-src 'unsafe-inline'", "expected": "script-src 'unsafe-inline'" },
  { "src": embedding_src, "csp": "totally invalid csp", "expected": null },
  ...
];

tests.forEach(test => {
  async_test(t => assert_embedding_csp(t, test.src, test.csp, test.expected));
});
```